gems/smithy-client/lib/smithy-client/api_key_signer.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    # Signs requests with the ApiKey identity.
+    class ApiKeySigner
+      def initialize(options = {})
+        @name = options[:name]
+        @in = options[:in]
+        @scheme = options[:scheme]
+      end
```

Comment:
Are these intended to be public interfaces - if so, we should include more documentation about init options?  Here, other signers and auth scheme.